marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/auth/AuthenticationFilter.java

```diff
@@ -96,6 +96,10 @@ protected AuthenticationToken authenticateUser(String username, String password)
                 if ("Failed Auth".equals(fre.getServerMessage()) || "Failed Auth".equals(fre.getServerStatus())) {
                     throw new BadCredentialsException("Unauthorized");
                 }
+                //In case user can't read the getAuthorities.sjs module
+                if(fre.getServerStatusCode() == 404 && fre.getMessage() != null && fre.getMessage().contains("Not Found")){
+                    throw new InsufficientAuthenticationException("User doesn't have necessary privileges to access Hub Central");
```

Comment:
@rjrudin ,
Currently, if a user logs in and doesn't have necessary authorities, we are throwing InsufficientAuthenticationException (line 110 AuthenticationFilter.java). That's why I did the same. Should that be changed as well ?

Reply:
Yeah, I think AccessDeniedException is the natural exception to throw here. The Spring javadocs state that the Insufficient exception is intended for when the method of authentication used is not sufficient : "AccessDecisionVoters will typically throw this exception if they are dissatisfied with the level of the authentication, such as if performed using a remember-me mechanism or anonymously. The ExceptionTranslationFilter will then typically cause the AuthenticationEntryPoint to be called, allowing the principal to authenticate with a stronger level of authentication."

Reply:
I've changed it so that if a user who successfully logs in and doesn't have sufficient privileges,  we don't throw "InsufficientAuthenticationException".